From 93e5a20f9daaa86b71156ca53c83ae955ea58331 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 27 Oct 2021 16:31:49 +0200
Subject: [PATCH 2/6] Fix davfs mount

Put quotes in correct places where davfs expects them, escape
potentially dangerous characters like spaces in passwords and make sure
the final secrets file have permissions required by davfs.
---
 schnapps.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/schnapps.sh b/schnapps.sh
index fcde10d..2d05523 100755
--- a/schnapps.sh
+++ b/schnapps.sh
@@ -634,9 +634,10 @@ webdav_mount() {
     mk_tmp_dir
     cat > "$TEMP_DIR"/dav-config << EOF
 use_locks 0
-secrets "$TEMP_DIR"/dav-secret
+secrets "$TEMP_DIR/dav-secret"
 EOF
-    echo "$FINAL_REMOTE_URL $REMOTE_USER $REMOTE_PASS" > "$TEMP_DIR"/dav-secret
+    echo "$TMP_RMT_MNT_DIR \"$REMOTE_USER\" \"$REMOTE_PASS\"" > "$TEMP_DIR"/dav-secret
+    chmod 0600 "$TEMP_DIR"/dav-secret
     mount.davfs "$FINAL_REMOTE_URL" "$TMP_RMT_MNT_DIR" -o dir_mode=0700,file_mode=0600,uid=root,gid=root,conf="$TEMP_DIR"/dav-config || die "Can't access remote filesystem"
 }
 
-- 
2.34.1

